src/Roave/SecurityAdvisories/Version.php

```diff
@@ -37,85 +38,90 @@ public static function fromString(string $version) : self
+        return version_compare($first, $second, '==');
     }
 
-    /**
-     * Compares two versions and sees if this one is greater than the given one
-     *
-     * @todo may become a simple array comparison (if PHP supports it)
-     *
-     * @param Version $other
-     *
-     * @return bool
-     */
     public function isGreaterThan(self $other) : bool
     {
-        foreach ($other->versionNumbers as $index => $otherVersion) {
-            $thisVersion = $this->versionNumbers[$index] ?? 0;
-
-            if ($thisVersion === $otherVersion) {
-                continue;
-            }
+        [$first, $second] = $this->normalizeVersions($other);
+        return version_compare($first, $second, '>');
+    }
 
-            return $thisVersion > $otherVersion;
-        }
+    public function isGreaterOrEqualThan(self $other) : bool
+    {
+        [$first, $second] = $this->normalizeVersions($other);
+        return version_compare($first, $second, '>=');
+    }
 
-        return (bool) array_filter(array_slice($this->versionNumbers, count($other->versionNumbers)));
+    public function toString()
+    {
+        return $this->version;
     }
 
     /**
-     * Compares two versions and sees if this one is greater or equal than the given one
-     *
-     * @todo may become a simple array comparison (if PHP supports it)
-     *
-     * @param Version $other
-     *
-     * @return bool
+     * Here we need to append zeroes so comparison will work correctly
      */
-    public function isGreaterOrEqualThan(self $other) : bool
+    private function normalizeVersions(Version $other): array
     {
-        foreach ($other->versionNumbers as $index => $otherVersion) {
-            $thisVersion = $this->versionNumbers[$index] ?? 0;
-
-            if ($thisVersion === $otherVersion) {
-                continue;
-            }
-
-            return $thisVersion > $otherVersion;
+        [$versionA, $tailA] = $this->stabilizeVersion($this->version);
+        [$versionB, $tailB] = $this->stabilizeVersion($other->version);
+
+        [$countA, $countB, $diff] = $this->getVersionStats($versionA, $versionB);
+        switch ($countA <=> $countB) {
+            case -1:
+                return [
+                    $versionA.str_repeat('.0', $diff).$tailA,
+                    $versionB.$tailB
+                ];
+            case 0;
+                return [
+                    $versionA.$tailA,
+                    $versionB.$tailB
+                ];
+            case 1;
+                return [
+                    $versionA.$tailA,
+                    $versionB.str_repeat('.0', $diff).$tailB
+                ];
         }
-
-        return true;
     }
 
-    public function getVersion() : string
+    /**
+     * Split version string representation in two parts - stable and stability tail,
+     * if no tail is present return null
+     */
+    private function stabilizeVersion(string $version) : array
```

Comment:
> Split version string representation in two parts - stable and stability tail

So, maybe: `splitVersionIntoStableAndStability()`?